v1/plugin/bool.go

```diff
@@ -19,44 +19,16 @@ limitations under the License.
 
 package plugin
 
-import "fmt"
-
-// boolRule defines a type to contain Bool specific rule data.
-type boolRule struct {
-	Key        string
-	Default    bool
-	HasDefault bool
-	Required   bool
-}
+import "github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/rpc"
 
-type boolRuleOpt func(*boolRule)
+type boolRuleOpt func(*rpc.BoolRule)
 
 // SetDefaultBool Allows easy setting of the Default value for an boolRule.
 // Usage:
 //		NewBoolRule(key, req, config.SetDefaultBool(default))
 func SetDefaultBool(in bool) boolRuleOpt {
-	return func(i *boolRule) {
+	return func(i *rpc.BoolRule) {
```

Comment:
I'm curious what you think about potentially merging the type files into one since they no longer contain structs.